ui/app/components/clients/dashboard.js

```diff
@@ -91,39 +107,73 @@ export default class Dashboard extends Component {
     }
     return this.args.model.activity.responseTimestamp;
   }
+  // HELPERS
+  areArraysTheSame(a1, a2) {
+    return (
+      a1 === a2 ||
+      (a1 !== null &&
+        a2 !== null &&
+        a1.length === a2.length &&
+        a1
+          .map(function (val, idx) {
+            return val === a2[idx];
+          })
+          .reduce(function (prev, cur) {
+            return prev && cur;
+          }, true))
+    );
+  }
 
+  // ACTIONS
   @action
   async handleClientActivityQuery(month, year, dateType) {
     if (dateType === 'cancel') {
       return;
     }
-    // dateType is either startTime or endTime
-    let monthIndex = this.arrayOfMonths.indexOf(month);
+    // clicked "Current Billing period" in the calendar widget
+    if (dateType === 'reset') {
+      this.startTimeRequested = this.args.model.startTimeFromLicense;
+      this.endTimeRequested = null;
+    }
+    // clicked "Edit" Billing start month in Dashboard which opens a modal.
     if (dateType === 'startTime') {
-      this.startTime = formatRFC3339(new Date(year, monthIndex));
-      this.endTime = null;
+      let monthIndex = this.arrayOfMonths.indexOf(month);
+      this.startTimeRequested = [year.toString(), monthIndex]; // ['2021', 0] (e.g. January 2021) // TODO CHANGE TO ARRAY
+      this.endTimeRequested = null;
     }
+    // clicked "Custom End Month" from the calendar-widget
     if (dateType === 'endTime') {
-      // this month comes in as an index
-      this.endTime = formatRFC3339(new Date(year, month));
+      // use the currently selected startTime for your startTimeRequested.
+      this.startTimeRequested = this.startTimeFromResponse;
+      this.endTimeRequested = [year.toString(), month];
     }
-    try {
-      let response = await this.adapter.queryClientActivity(this.startTime, this.endTime);
-      // resets the endTime to what is returned on the response
-      this.endTime = response.data.end_time;
 
+    try {
+      let response = await this.store.queryRecord('clients/activity', {
+        start_time: this.startTimeRequested,
+        end_time: this.endTimeRequested,
+      });
+      if (!response) {
+        // this.endTime will be null and use this to show EmptyState message on the template.
+        return;
+      }
+      this.startTimeFromResponse = response.formattedStartTime;
+      this.endTimeFromResponse = response.formattedEndTime;
+
+      if (!this.areArraysTheSame(this.startTimeFromResponse, this.startTimeRequested)) {
```

Comment:
Perhaps I'm misreading this but should this read "We only have data from `this.startTimeFromResponse`" and not `this.startTimeDisplay` since we're comparing the user's request to what was actually returned from the API?